corehq/motech/dhis2/tests/test_views.py

```diff
@@ -0,0 +1,177 @@
+            'column': datavalue_map.column,
+            'data_element_id': datavalue_map.data_element_id,
+            'category_option_combo_id': datavalue_map.category_option_combo_id,
+            'comment': new_comment,
+            'action': 'update'
+        }
+
+        url_kwargs = {
+            'domain': self.domain.name,
+            'pk': dataset_map.pk
+        }
+        response = self.client.post(reverse(self.view_path, kwargs=url_kwargs), data)
+
+        self.assertEqual(response.status_code, 404)
+
+    def test_flag_not_enabled(self):
+        (dataset_map, datavalue_map) = (self.dataset_map, self.data_value_map)
+        new_comment = 'Testy comment one'
+
+        data = {
+            'id': datavalue_map.id,
+            'column': datavalue_map.column,
+            'data_element_id': datavalue_map.data_element_id,
+            'category_option_combo_id': datavalue_map.category_option_combo_id,
+            'comment': new_comment,
+            'action': 'update'
+        }
+
+        url_kwargs = {
+            'domain': self.domain.name,
+            'pk': dataset_map.pk
+        }
+        response = self.client.post(reverse(self.view_path, kwargs=url_kwargs), data)
+
+        self.assertEqual(response.status_code, 404)
+
+    @flag_enabled('DHIS2_INTEGRATION')
+    def test_update_ok(self):
+        (dataset_map, datavalue_map) = (self.dataset_map, self.data_value_map)
+        new_comment = 'Testy comment one'
+
+        data = {
+            'id': datavalue_map.id,
+            'column': datavalue_map.column,
+            'data_element_id': datavalue_map.data_element_id,
+            'category_option_combo_id': datavalue_map.category_option_combo_id,
+            'comment': new_comment,
+            'action': 'update'
+        }
+
+        url_kwargs = {
+            'domain': self.domain.name,
+            'pk': dataset_map.pk
+        }
+
+        response = self.client.post(reverse(self.view_path, kwargs=url_kwargs), data)
+
+        self.assertEqual(response.status_code, 200)
+
+        datavalue_map.refresh_from_db()
+        self.assertEqual(datavalue_map.comment, new_comment)
+
+    @flag_enabled('DHIS2_INTEGRATION')
+    def test_invalid_data_element_id_update(self):
+        (dataset_map, datavalue_map) = (self.dataset_map, self.data_value_map)
+        invalid_data_element_id = "123"  # Must start with letter and be 11 chars long
+
+        data = {
+            'id': datavalue_map.id,
+            'column': datavalue_map.column,
+            'data_element_id': invalid_data_element_id,
+            'category_option_combo_id': datavalue_map.category_option_combo_id,
+            'comment': datavalue_map.comment,
+            'action': 'update'
+        }
+
+        url_kwargs = {
+            'domain': self.domain.name,
+            'pk': dataset_map.pk
+        }
+
+        response = self.client.post(reverse(self.view_path, kwargs=url_kwargs), data)
+        self.assertEqual(response.status_code, 200)
```

Comment:
If it's quick to do, I'd be curious to know what the response looks like. But good to go if it'll long.